app/gui/src/project-view/components/GraphEditor/widgets/WidgetSelfAccessChain.vue

```diff
@@ -27,8 +27,22 @@ const { displayedIcon } = useDisplayedIcon(graph.db, toRef(tree, 'externalId'),
 const iconInput = computed(() => {
   const lhs = props.input.value.lhs
   if (!lhs) return
-  const input = WidgetInput.WithPort(WidgetInput.FromAst(lhs))
-  input[DisplayIcon] = { icon: displayedIcon.value, showContents: showFullAccessChain.value }
+  let expression = lhs
+  let portId = lhs.id
+  // Unwrap the Group if present.
+  if (expression instanceof Ast.Group && expression.expression) {
+    expression = expression.expression
+    portId = expression.id
+  }
+  if (expression instanceof Ast.TypeAnnotated) {
```

Comment:
Again, there might be multiple groups. And annotations perhaps too?

Reply:
Here I’m using `unwrapGroups` now. I tried looking into nested type annotations, but the engine does not support them, as they seem to be recognized as “type operator chains”. I feel this implementation is robust against the most obvious syntactic variations that the user can reasonably come up with.

Reply:
And of course, it is not clear how nested type annotations should translate into a self-access chain anyway, so let’s leave it to the future selves.